### PR TITLE
First step with TF_DEBUG codes

### PR DESCRIPTION
**Pull request by dedoardo**
Issuing a possibly useful warning through its own debug code rather than a generic warning/error. This can be controlled through the environment variable `TF_DEBUG=HDCYCLES_MESH` or `TF_DEBUG=HDCYCLES_*` in the future.

In the future non fatal application-level messages should go through our custom codes and ideally the USD/Hydra warnings should disappear through better use of the API.

### CODE REVIEW

File: plugin/hdCycles/debug_codes.h [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied,
+//  including without limitation, as related to merchantability and fitness
+//  for a particular purpose.
+//
+//  In no event shall any copyright holder be liable for any damages of any kind
+//  arising from the use of this software, whether in contract, tort or otherwise.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+#pragma once
```
Comment (bareya) [inline plugin/hdCycles/debug_codes.h:19]: Unfortunately, `pragma once` is not part of the standard. It is supported by `gcc`, but I think we should stick to the standard `#ifndef, #define` as we do everywhere else in the project.

File: plugin/hdCycles/mesh.cpp [hunk 2]
```diff
@@ -533,9 +534,9 @@ HdCyclesMesh::_PopulateNormals(HdSceneDelegate* sceneDelegate, const SdfPath& id
 
     HdInterpolation interpolation = HdInterpolationCount;
     if (!GetPrimvarInterpolation(interpolation)) {
-        // TODO: Should we autogenerate normals or let Cycles generate them?
-        // Let's shoot a warning for now
-        //TF_WARN("Failed to find interpolation for normals for: %s", id.GetText());
+        m_cyclesMesh->add_vertex_normals();
+        TF_INFO(HDCYCLES_MESH)
```
Comment (bareya) [inline plugin/hdCycles/mesh.cpp:538]: Is this info printed every time? Maybe it should be only during debug? I don't think we have to notify users about this detail.

File: plugin/hdCycles/debug_codes.h (hunk 1 above)
Comment (dedoardo) [reply plugin/hdCycles/debug_codes.h]: Fixed

File: plugin/hdCycles/mesh.cpp (hunk 2 above)
Comment (dedoardo) [reply plugin/hdCycles/mesh.cpp:538]: It's printed only if you set the env variable TF_DEBUG to match HDCYCLES_*.
Is it so common to author models without any normals?

Comment (boberfly) [reply plugin/hdCycles/mesh.cpp:538]: Yeah it's common to not have normals if the mesh was intended to be a subdivision surface. The expected behaviour would be normals that get generated by the renderer, all smooth. Saves storing normals to disk if they can be auto-generated.

Comment (dedoardo) [reply plugin/hdCycles/mesh.cpp:538]: @boberfly The case of subdivision surfaces is [already handled earlier in the function](https://github.com/tangent-opensource/hdcycles/blob/fix/populate_normals_warnings/plugin/hdCycles/mesh.cpp#L502). This is for non-subdivision meshes which don't have normals.

Comment (bareya) [reply plugin/hdCycles/mesh.cpp:538]: @boberfly When mesh us subdivided, normals are taken from limit surface.

Comment (boberfly) [reply plugin/hdCycles/mesh.cpp:538]: Hmm, I guess for regular geometry the intended behaviour would be that the renderer would still generate smooth normals. I'm not sure if Cycles already does this.

Comment (dedoardo) [reply plugin/hdCycles/mesh.cpp:538]: Yes, it does it internally before committing the geometry. Now it's explicit and notifies the user.

**Pull request merged**